pkg/pgclient/client.go

```diff
@@ -117,44 +117,48 @@ func (cfg *Config) GetConnectionStr() string {
 		cfg.host, cfg.port, cfg.user, cfg.database, cfg.password, cfg.sslMode)
 }
 
-func (cfg *Config) GetNumConnections() (min int, max int, perProc int, err error) {
+func (cfg *Config) GetNumConnections() (min int, max int, numCopiers int, err error) {
 	maxProcs := runtime.GOMAXPROCS(-1)
 	if cfg.ConnectionsPerProc < 1 {
 		return 0, 0, 0, fmt.Errorf("invalid number of connections-per-proc %v, must be at least 1", cfg.ConnectionsPerProc)
 	}
-	perProc = cfg.ConnectionsPerProc
-	dbMaxConnections := cfg.MaxConnections
-	if dbMaxConnections < 1 {
+	perProc := cfg.ConnectionsPerProc
+	max = cfg.MaxConnections
+	if max < 1 {
 		conn, err := pgx.Connect(context.Background(), cfg.GetConnectionStr())
 		if err != nil {
 			return 0, 0, 0, err
 		}
 		defer func() { _ = conn.Close(context.Background()) }()
 		row := conn.QueryRow(context.Background(), "SHOW max_connections")
-		err = row.Scan(dbMaxConnections)
+		err = row.Scan(&max)
 		if err != nil {
 			return 0, 0, 0, err
 		}
-		if dbMaxConnections <= 1 {
+		if max <= 1 {
 			log.Warn("msg", "database can only handle 1 connection")
-			return dbMaxConnections, dbMaxConnections, 1, nil
+			return 1, 1, 1, nil
 		}
-		dbMaxConnections = int(0.8 * float32(dbMaxConnections))
+		// we try to only use 80% the database connections
+		max = int(0.8 * float32(max))
```

Comment:
shouldn't this be the min(int(0.8 * float32(max)), perProc*maxProcs)?

Reply:
or maybe this should be below the if?

Reply:
I don't think so; if the user wants more connections for e.g. readers, it should be allowed.

Reply:
So then the `db-connections-per-proc` is only used for calculating the numCopiers? Is that right? If so we should make that explicit in the description of the option